packages/appsync-modelgen-plugin/src/visitors/appsync-dart-visitor.ts

```diff
@@ -718,24 +658,55 @@ export class AppSyncModelDartVisitor<
     );
   }
 
+  protected generateCopyWithModelFieldValuesMethod(model: CodeGenModel, declarationBlock: DartDeclarationBlock): void {
+    // copyWithModelFieldValues
+    const writableFields = this.getWritableFields(model, this.isCustomPKEnabled());
+    const copyParameters = writableFields.map(
+      field => `ModelFieldValue<${this.getNativeType(field)}${field.isNullable ? '?' : ''}>? ${this.getFieldName(field)}`,
+    );
+    const copyParameterStr = `{\n${indentMultiline(copyParameters.join(',\n'))}\n}`;
+    declarationBlock.addClassMethod(
+      'copyWithModelFieldValues',
+      this.getModelName(model),
+      writableFields.length ? [{ name: copyParameterStr }] : undefined,
+      [
+        `return ${this.getModelName(model)}${this.config.isTimestampFieldsAdded ? '._internal' : ''}(`,
+        indentMultiline(
+          `${this.getWritableFields(model, false)
```

Comment:
Why using the `${this.getWritableFields(model, false)` instead of `const writableFields = this.getWritableFields(model, this.isCustomPKEnabled());` here?

Reply:
`this.getWritableFields(model, this.isCustomPKEnabled())` gets fields that are not a part of CPK and is not readonly.
`this.getWritableFields(model, false)` gets fields that are not readonly. When create a new instance, all fields are needed.

Reply:
Ah I thought it was different from `copyWith`. The param is default as false so they are the same.